File: lyedu-api-python/routers/certificate.py
# -*- coding: utf-8 -*-
"""证书颁发规则路由，与 Java CertificateController 对应"""
from typing import Optional

from fastapi import APIRouter
from pydantic import BaseModel

from common.result import error_result, success
from services import certificate_service

router = APIRouter(prefix="/certificate", tags=["certificate"])


class CertificateRequest(BaseModel):
    templateId: Optional[int] = None
    name: str = ""
    sourceType: str = ""
    sourceId: Optional[int] = None
    sort: Optional[int] = 0
    status: Optional[int] = 1


@router.get("/list")
def list_all():
    return success(certificate_service.list_all())


@router.get("/{cert_id}")
def get_by_id(cert_id: int):
    c = certificate_service.get_by_id(cert_id)
    if not c:
        return error_result((404, "资源不存在"))
    return success(c)


@router.post("")
def create(body: CertificateRequest):
    if body.templateId is None or body.sourceId is None:
        return error_result((400, "模板ID和来源ID不能为空"))
    entity = {
        "templateId": body.templateId,
        "name": body.name,
        "sourceType": body.sourceType,
        "sourceId": body.sourceId,
        "sort": body.sort or 0,
        "status": body.status or 1,
    }
    rid = certificate_service.save(entity)
    return success(rid)


@router.put("/{cert_id}")
def update(cert_id: int, body: CertificateRequest):
    c = certificate_service.get_by_id(cert_id)
    if not c:
        return error_result((404, "资源不存在"))
    entity = {
        "id": cert_id,
        "templateId": body.templateId,
        "name": body.name,
        "sourceType": body.sourceType,
        "sourceId": body.sourceId,
        "sort": body.sort or 0,
        "status": body.status or 1,
    }
    certificate_service.update(entity)
    return success(None)


@router.delete("/{cert_id}")
def delete(cert_id: int):
    c = certificate_service.get_by_id(cert_id)
    if not c:
        return error_result((404, "资源不存在"))
    certificate_service.delete(cert_id)
    return success(None)

File: lyedu-api/src/main/java/com/lyedu/entity/Certificate.java
package com.lyedu.entity;

import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * 证书颁发规则实体（关联考试/任务，合格后颁发）
 *
 * @author LyEdu Team
 */
@Data
@EqualsAndHashCode(callSuper = true)
public class Certificate extends BaseEntity {

    /** 证书模板ID */
    private Long templateId;
    /** 证书名称（如：Java 考试合格证） */
    private String name;
    /** 来源类型：exam-考试，task-任务 */
    private String sourceType;
    /** 来源ID（考试ID或任务ID） */
    private Long sourceId;
    /** 排序 */
    private Integer sort;
    /** 状态：0-禁用，1-启用 */
    private Integer status;
}
